server/src/test/ZetaSqlWrapper.spec.ts

```diff
@@ -172,12 +176,38 @@ describe('ZetaSqlWrapper', () => {
 
   it('register should unregister deleted column', () => {
     // arrange
+    const zetaSqlWrapper = new ZetaSqlWrapper(mock(DbtRepository), mock(BigQueryClient));
     const tableDefinition = new TableDefinition([PROJECT_ID, DATA_SET, TABLE]);
     tableDefinition.columns = [ONE_COLUMN, TWO_COLUMN];
-    registerTable(tableDefinition);
+    registerTable(zetaSqlWrapper, tableDefinition);
 
     // act, assert
     tableDefinition.columns.pop();
     shouldRegisterTable(tableDefinition, TABLE, [ONE_COLUMN_NAME], DATA_SET, PROJECT_ID);
   });
+
+  it('registerUdf should register UDF', () => {
+    const zetaSqlWrapper = new ZetaSqlWrapper(mock(DbtRepository), mock(BigQueryClient));
+    const udf: Udf = {
+      nameParts: ['udfs', 'func'],
+      arguments: [
+        {
+          name: 'arg',
+          type: {
+            typeKind: TypeKind.TYPE_INT64,
+          },
+        },
+      ],
+      returnType: {
+        typeKind: TypeKind.TYPE_INT64,
+      },
+    };
+
+    const rootCatalog = registerUdf(zetaSqlWrapper, udf);
+
+    const udfsCatalog = rootCatalog.catalog?.find(c => c.name === 'udfs');
+    assert.ok(udfsCatalog);
+    const func = udfsCatalog.customFunction?.find(c => c.namePath && arraysAreEqual(c.namePath, ['udfs', 'func']));
+    assert.ok(func);
```

Comment:
Do we need to check registered arguments, return type?

Reply:
Yes, added asserts.